### PR TITLE
Added missing likely locales

### PR DESCRIPTION
**Pull request by ehoogerbeets**
### Checklist

* [x] At least one test case is included for this feature or bug fix.
* [x] `ReleaseNotes` has added or is not needed.
* [x] This PR has passed all of the test cases on `Nodejs` and `Chrome Browser`.
* [x] This PR has passed all of the test cases on `QT/QML`.
* [ ] This is an API breaking change.
* [ ] Requires a major version change.

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Links
[//]: # (Related issues, references)

### CODE REVIEW

**Comment by ehoogerbeets** [comment]
@gouniLee view this PR with the "Hide whitespace changes" turned on or else it will be much larger than it should be. (I changed the testlocalematch.json from dos to unix line endings).

Also, add in any missing locales you find from your investigations and add unit tests for them.

**Comment by gouniLee** [comment]
@ehoogerbeets  I've added more locales to 'likelyLocalesAdditional.json` and unit test to verify them. Please review them.

File: js/data/locale/localematch.json [hunk 1]
```diff
@@ -5528,7 +5528,221 @@
         "tlh": "tlh-Piqd",
         "tlh-Piqd": "tlh-Piqd",
         "tlh-PIQD": "tlh-Piqd",
-        "Piqd-PIQD": "tlh-Piqd"
+        "Piqd-PIQD": "tlh-Piqd",
+        "ar-DJ": "ar-Arab-DJ",
+        "und-ar-DJ": "ar-Arab-DJ",
+        "ar-AR-DJ": "ar-Arab-DJ",
```
Comment (ehoogerbeets) [inline js/data/locale/localematch.json]: Why are there bcp-47 codes that have two regions or two languages or two scripts listed here?

File: js/data/locale/localematch.json [hunk 2]
```diff
@@ -5528,7 +5528,221 @@
...
+        "en-EN-PH": "en-Latn-PH",
+        "Latn-EN-PH": "en-Latn-PH",
+        "en-PK": "en-Latn-PK",
+        "und-en-PK": "en-Latn-PK",
+        "en-EN-PK": "en-Latn-PK",
+        "Latn-EN-PK": "en-Latn-PK",
+        "en-PR": "en-Latn-PR",
+        "und-en-PR": "en-Latn-PR",
+        "en-EN-PR": "en-Latn-PR",
+        "Latn-EN-PR": "en-Latn-PR",
```
Comment (ehoogerbeets) [inline js/data/locale/localematch.json]: Capital "EN" is not even a valid ISO country code... why is it here?

File: js/data/locale/localematch.json [hunk 3]
```diff
@@ -5528,7 +5528,221 @@
...
+        "Latn-FR-DZ": "fr-Latn-DZ",
+        "fr-LB": "fr-Latn-LB",
+        "und-fr-LB": "fr-Latn-LB",
+        "fr-FR-LB": "fr-Latn-LB",
+        "Latn-FR-LB": "fr-Latn-LB",
+        "fr-RW": "fr-Latn-RW",
+        "und-fr-RW": "fr-Latn-RW",
+        "fr-FR-RW": "fr-Latn-RW",
+        "Latn-FR-RW": "fr-Latn-RW",
+        "und-ha-NG": "fr-Latn-NG",
```
Comment (ehoogerbeets) [inline js/data/locale/localematch.json]: I don't think French is spoken in Nigeria at all. This should map to ha-Latn-NG instead.

File: js/data/locale/localematch.json [hunk 4]
```diff
@@ -5528,7 +5528,221 @@
...
+        "uk-UK-UA": "uk-Cyrl-UA",
+        "Cyrl-UK-UA": "uk-Cyrl-UA",
+        "ur-IN": "ur-Arab-IN",
+        "und-ur-IN": "ur-Arab-IN",
+        "ur-UR-IN": "ur-Arab-IN",
+        "Arab-UR-IN": "ur-Arab-IN",
+        "und-ur-PK": "ur-Arab-PK",
+        "ur-UR-PK": "ur-Arab-PK",
+        "Arab-UR-PK": "ur-Arab-PK",
+        "und-uz-UZ": "uz-Latn-UA",
```
Comment (ehoogerbeets) [inline js/data/locale/localematch.json]: Uzbek is not spoken in Ukraine. You probably meant "uz-Latn-UZ" instead, right?

File: tools/cldr/likelyLocalesAdditional.json [hunk 5]
```diff
@@ -263,5 +263,62 @@
...
+    "sq-ME": "sq-Latn-ME",
+    "sv-FI": "sv-Latn-FI",
+    "sv-SE": "sv-Latn-SE",
+    "tr-AM": "tr-Latn-AM",
+    "tr-AZ": "tr-Latn-AZ",
+    "tr-CY": "tr-Latn-CY",
+    "uk-UA": "uk-Cyrl-UA",
+    "ur-IN": "ur-Arab-IN",
+    "ur-PK": "ur-Arab-PK",
+    "uz-UZ": "uz-Latn-UA"
```
Comment (ehoogerbeets) [inline tools/cldr/likelyLocalesAdditional.json]: uz-Latn-UZ

File: tools/cldr/likelyLocalesAdditional.json [hunk 6]
```diff
@@ -263,5 +263,62 @@
...
+    "ru-KG": "ru-Cyrl-KG",
+    "ru-UA": "ru-Cyrl-UA",
+    "sq-ME": "sq-Latn-ME",
+    "sv-FI": "sv-Latn-FI",
+    "sv-SE": "sv-Latn-SE",
+    "tr-AM": "tr-Latn-AM",
+    "tr-AZ": "tr-Latn-AZ",
+    "tr-CY": "tr-Latn-CY",
+    "uk-UA": "uk-Cyrl-UA",
+    "ur-IN": "ur-Arab-IN",
```
Comment (ehoogerbeets) [inline tools/cldr/likelyLocalesAdditional.json]: Should this be ur-Deva-IN ??

File: js/data/locale/localematch.json (hunk 1 above)
Comment (gouniLee) [reply js/data/locale/localematch.json]: hm.. It's weird. I only added locales to likelyLocalesAdditionsl.json form like (##-##) and regenerate localematch.json by running genlikely.js. That's not I expected form.

File: tools/cldr/likelyLocalesAdditional.json (hunk 6 above)
Comment (gouniLee) [reply tools/cldr/likelyLocalesAdditional.json]: No. Arab is correct. `Deva` is for `mr` and  `hi' language.

File: tools/cldr/likelyLocalesAdditional.json (hunk 5 above)
Comment (gouniLee) [reply tools/cldr/likelyLocalesAdditional.json]: I'll fix it

File: js/data/locale/localematch.json (hunk 1 above)
Comment (gouniLee) [reply js/data/locale/localematch.json]: It looks like  genlikelyloc.js` tool code, doing more work.

**Comment by ehoogerbeets** [comment]
The other branch for the getLikelyLocaleMinimal should go first, as there are conflicts with this branch and there are updates to the genlikelylocales that need to be merged in and the localematch.json regenerated with them.

**Comment by ehoogerbeets** [comment]
@gouniLee please re-review the changes from fixing the merge conflicts.

**Pull request merged**